lib/new_relic/agent/instrumentation/controller_instrumentation.rb

```diff
@@ -230,6 +231,8 @@ def already_added_transaction_tracer?(target, with_method_name)
 
         class TransactionNamer
           def self.name_for(txn, traced_obj, category, options = {})
+            return options[:transaction_name] if options[:transaction_name]
+
```

Comment:
- Enhance `NewRelic::Agent::Instrumentation::ControllerInstrumentation::TransactionNamer.name_for`
  to respect a `:transaction_name` key/value pair in the options hash when present.
  This allows for the permanent manual specification of a transaction name when
  desired.